admin/src/components/ReportsReviewTable/index.tsx

```diff
@@ -90,6 +91,20 @@ const ReportsReviewTable = ({
     );
   };
 
+  const handleItemSelectionChange = (selection, value) => {
+    if (isArray(selection)) {
+      onSelectionChange(value ? selection : [])
+    } else {
+      onSelectionChange([...selectedItems, selection].filter(_ => !value ? selection !== _ : value));
```

Comment:
```suggestion
      onSelectionChange([...selectedItems, selection].filter(item => value || selection !== item));
```
That's a bit more readable to me. Not 100% sure it's a better solution. Just food for thought.